syncstorage/db.go

```diff
@@ -928,15 +920,29 @@ func (d *DB) getBSOs(
 		}
 	}
 
-	nextOffset := 0
-	more := (totalRows > limit+offset)
-	if more {
-		nextOffset = offset + limit
+	var more bool
+	var nextOffset int
+
+	// if a limit was applied, determine if there are records beyond
+	// the upper bound and return the pertinent information
+	if limit > 0 {
+		var totalRows int
+		countQuery := "SELECT COUNT(1) NumRows FROM BSO " + where + " " + orderBy
```

Comment:
This could be an expensive extra query for large collections; a future optimization could be to use `LIMIT {{limit}} + 1` in your original query, and then set `more = true` if you managed to read more than `{{limit}}` items.  NBD right now though.